Deprecate uninstall --everything instead of removing it

thallgren opened the pull request:
## Description

The `telepresence uninstall` command was completely removed in 2.7.0
which may create some unnecessary churn for our users. The functionality
was moved in unmodified under `telepresence helm uninstall`. That was
a bit unfortunate, because `helm` isn't involved when uninstalling
agents. Also, it requires a `--everything` flag to actually remove
the traffic-manager.

This commit reinstates the `telepresence uninstall` command but
deprecates the `--everything` flag. It can still be used, but now it
delegates to `telepresence helm uninstall` and informs the user that
it's deprecated.

This commit also changes `telepresence helm uninstall` into a command
that just performs a `helm uninstall` of the traffic-manager.

A few sentences describing the overall goals of the pull request's commits.

 - [x] I made sure to update `./CHANGELOG.md`.
 - [x] My change is adequately tested.

Commit:
Deprecate uninstall --everything instead of removing it

The `telepresence uninstall` command was completely removed in 2.7.0
which may create some unnecessary churn for our users. The functionality
was moved in unmodified under `telepresence helm uninstall`. That was
a bit unfortunate, because `helm` isn't involved when uninstalling
agents. Also, it requires a `--everything` flag to actually remove
the traffic-manager.

This commit reinstates the `telepresence uninstall` command but
deprecates the `--everything` flag. It can still be used, but now it
delegates to `telepresence helm uninstall` and informs the user that
it's deprecated.

This commit also changes `telepresence helm uninstall` into a command
that just performs a `helm uninstall` of the traffic-manager.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

njayp commented on pkg/install/helm/install.go:
```diff
@@ -233,22 +234,16 @@ func EnsureTrafficManager(ctx context.Context, configFlags *genericclioptions.Co
-		} else {
-			return errcat.User.Newf("traffic manager version %q is already installed, use the '--upgrade' flag to replace it", releaseVer(existing))
-		}
+	case releaseVer(existing) == strings.TrimPrefix(client.Version(), "v"):
+		dlog.Infof(ctx, "traffic manager version %q is up to date", releaseVer(existing))
```
what if the user wants to update the helm chart with a different --values file?

njayp commented on pkg/client/userd/trafficmgr/traffic_manager.go:
```diff
@@ -351,8 +351,7 @@ func connectCluster(c context.Context, cr *rpc.ConnectRequest) (*k8s.Cluster, er
+func DeleteManager(ctx context.Context, req *rpc.HelmRequest) error {
 	cr := req.GetConnectRequest()
 	if cr == nil {
 		dlog.Info(ctx, "Connect_request in install_request was nil, using defaults")
```
nit log update

njayp commented on pkg/client/userd/trafficmgr/traffic_manager.go:
```diff
@@ -364,14 +363,25 @@ func EnsureManager(ctx context.Context, req *rpc.InstallRequest) error {
+	// seg guard
+	cr := req.GetConnectRequest()
+	if cr == nil {
+		dlog.Info(ctx, "Connect_request in install_request was nil, using defaults")
```
nit log update

njayp commented on pkg/client/userd/trafficmgr/traffic_manager.go:
```diff
@@ -396,15 +406,14 @@ func connectMgr(c context.Context, cluster *k8s.Cluster, installID string, svc S
-	if err != nil {
-		return nil, stacktrace.Wrap(err, "new installer")
+	dlog.Debug(c, "checking that traffic-manager exists")
+	existing, _, isManagerErr := helm.IsTrafficManager(c, cluster.ConfigFlags, cluster.GetManagerNamespace())
+	if isManagerErr == nil && existing == nil {
+		return nil, errcat.User.New("traffic manager not found, if it is not installed, please run 'telepresence helm install'")
 	}
```
this was my first approach, but a test failed, i think it was because on one test we have some missing permissions, but we want the install to succeed, that's why I still attempted the install even if I got nil, nil. If I missed additional changes, pls disregard

thallgren replied:
Yeah, I also changed the `IsTrafficManager` so that it actually returns an error if it runs into permission problems.

njayp commented on rpc/connector/connector.proto:
```diff
@@ -196,13 +196,19 @@ message IngressInfos {
 }
 
-message InstallResult {
+message Result {
```
nit can this be `HelmResult`

thallgren replied:
Not really, because it's also used by the `Uninstall` method.

njayp reviewed (changes requested):
initial comments, I will look at it again in a bit

Commit:
Changes in response to code-review

- Fix wrong struct name in log statements
- Don't assume install is up-to-date just because version match

Signed-off-by: Thomas Hallgren <thomas@datawire.io>
